metsim/disaggregate.py

```diff
@@ -75,17 +75,21 @@ def temp(df_daily, df_disagg, t_t_min, t_t_max, ts):
     temp = np.append(np.insert(temp, 0, temp[0:2]), temp[-2:])
 
     # Interpolate the values
-    interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
-    temps = interp(ts * np.arange(0, len(df_disagg.index)))
+    try:
+        interp = scipy.interpolate.PchipInterpolator(time, temp, extrapolate=True)
+        temps = interp(ts * np.arange(0, len(df_disagg.index)))
+    except ValueError:
+        temps = np.full(len(df_disagg.index), np.nan)
+
```

Comment:
I think this is too deep in the code to be handling domain/mask issues.  Ideally, we just iterate over the cells that we care about.